### PR TITLE
fix: add libatomic1 for node v25, node-gyp when using pnpm

### PR DESCRIPTION
**Pull request by iloveitaly**
* node v25 requires libatomic
  * we did not have integration tests for edge node versions, so this wasn't caught
  * adding tests for edge node and edge pnpm/npm
  * also adding a suite of integration tests for packages which require compilation (node-gyp is failing is some pnpm cases)
* additionally, if you install pnpm via mise node-gyp is not installing, causing builds to fail. I partially fixed this in this PR, although it will need some additional work for pnpm edge cases.
  * Notably: this work discovered a bug where if you corepack-install pnpm (no corepack) it fails to run since the shim for it is removed/not kept. Will handle that in a separate PR.

References:

* https://station.railway.com/questions/after-the-new-railpack-0-12-0-we-are-get-b27668da
* https://discord.com/channels/713503345364697088/1440071711239176192

### CODE REVIEW

File: core/providers/node/package_manager.go [hunk 1]
```diff
@@ -108,6 +108,15 @@ func (p PackageManager) installDeps(ctx *generate.GenerateContext, install *gene
...
 		}
 	case PackageManagerPnpm:
+		// pnpm (standalone) does not bundle node-gyp like npm does, so we must install it globally
+		// to support packages with native dependencies (e.g., better-sqlite3, bcrypt, etc.)
+		// Set PNPM_HOME so pnpm can create a global bin directory for node-gyp
+		install.AddEnvVars(map[string]string{
+			"PNPM_HOME": "/pnpm",
+		})
+		install.AddPaths([]string{"/pnpm"})
+		install.AddCommand(plan.NewExecCommand("pnpm add -g node-gyp"))
```
[the remark this answers is not in the document]
Comment (iloveitaly) [reply core/providers/node/package_manager.go]: @coffee-cup I'll check and add an integration test around this. My gut is it is only required when pnpm is installed via mise.
